lib/client/api_login_test.go

```diff
@@ -193,6 +197,27 @@ func TestTeleportClient_Login_local(t *testing.T) {
 			solveWebauthn: solvePwdless,
 			authConnector: constants.PasswordlessConnector,
 		},
+		{
+			name:                 "auth=local but use passwordless if was registered",
+			secondFactor:         constants.SecondFactorOptional,
+			inputReader:          prompt.NewFakeReader(), // no inputs
+			solveWebauthn:        solvePwdless,
+			authConnector:        constants.LocalConnector,
+			isPswdlessRegistered: true,
+		},
+		{
+			name:         "auth=local, passwordless registered but using cross-platform att",
+			secondFactor: constants.SecondFactorOptional,
+			inputReader: prompt.NewFakeReader().
+				AddString(password).
+				AddReply(func(ctx context.Context) (string, error) {
+					panic("this should not be called")
+				}),
+			solveWebauthn:        solveWebauthn,
+			authConnector:        constants.LocalConnector,
+			isPswdlessRegistered: true,
+			attachment:           wancli.AttachmentCrossPlatform,
+		},
```

Comment:
Do the new tests break with the old code?

Do they break if we remove the authenticator attachment if from the new code?

Reply:
yes and yes

Reply:
Nice, thanks.